package/package.sh

```diff
@@ -206,7 +233,54 @@ function package {
     popd
 }
 
+function package_tar_sh {
+    exec_file=$build_dir/nebula-$version$package_name.sh
+
+    echo "Creating self-extractable package $exec_file"
+    cat > $exec_file <<EOF
+#! /usr/bin/env bash
+set -e
+hash xz &> /dev/null || { echo "xz: Command not found"; exit 1; }
+[[ \$# -ne 0 ]] && prefix=\$(echo "\$@" | sed 's;.*--prefix=(\S*).*;\1;p' -rn)
+prefix=\${prefix:-/usr/local/nebula}
+mkdir -p \$prefix
+[[ -w \$prefix ]] || { echo "\$prefix: No permission to write"; exit 1; }
+archive_offset=\$(awk '/^__start_of_archive__$/{print NR+1; exit 0;}' \$0)
+tail -n+\$archive_offset \$0 | tar --no-same-owner --numeric-owner -xJf - -C \$prefix
+daemons=(metad graphd storaged)
+for daemon in \${daemons[@]}
+do
+    if [[ ! -f \$prefix/etc/nebula-\$daemon.conf ]] && [[ -f \$prefix/etc/nebula-\$daemon.conf.default ]]; then
+        cp \$prefix/etc/nebula-\$daemon.conf.default \$prefix/etc/nebula-\$daemon.conf
+        chmod 644 \$prefix/etc/nebula-\$daemon.conf
+    fi
+done
+echo "Nebula Graph has been installed to \$prefix"
+exit 0
+__start_of_archive__
+EOF
+    pushd $install_dir
+    tar -cJf - * >> $exec_file
+    chmod 0755 $exec_file
+    echo "####### target package file is $exec_file"
+    popd
+}
 
 # The main
-build $version $enablesanitizer $static_sanitizer $build_type $branch
-package $strip_enable
+gen_package_name
+
+if [[ $package_type == "auto" ]]; then
+    build
+    package
+else
+    install_dir=${build_dir}/install
+    build $version $enablesanitizer $static_sanitizer $buildtype
```

Comment:
these arguments could be removed and need not to process the inputs in the `build` function.